webapp/src/components/SmartTagsTable.tsx

```diff
@@ -0,0 +1,476 @@
+import { ArrowForward } from "@mui/icons-material";
+import {
+  Box,
+  FormControlLabel,
+  Link,
+  MenuItem,
+  Select,
+  Switch,
+  TableSortLabel,
+  TableSortLabelProps,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+import noData from "assets/void.svg";
+import Loading from "components/Loading";
+import { motion } from "framer-motion";
+import React from "react";
+import { Link as RouterLink } from "react-router-dom";
+import { getOutcomeCountPerFilterEndpoint } from "services/api";
+import {
+  AvailableDatasetSplits,
+  DatasetSplitName,
+  OutcomeCountPerFilterValue,
+} from "types/api";
+import { QueryPipelineState } from "types/models";
+import {
+  ALL_OUTCOMES,
+  OUTCOME_COLOR,
+  OUTCOME_PRETTY_NAMES,
+  SmartTagFamily,
+  SMART_TAG_FAMILIES,
+  SMART_TAG_FAMILY_ICONS,
+  SMART_TAG_FAMILY_PRETTY_NAMES,
+  UNKNOWN_ERROR,
+} from "utils/const";
+import { formatRatioAsPercentageString } from "utils/format";
+import { constructSearchString } from "utils/helpers";
+import DatasetSplitToggler from "./Controls/DatasetSplitToggler";
+
+type SortBy = "filterValue" | "utteranceCount" | "accuracy" | SmartTagFamily;
+type Row = {
+  filterValue: string;
+  utteranceCount: number;
+  accuracy: number;
+  outcomeCountPerSmartTagFamily: Partial<
+    Record<SmartTagFamily, OutcomeCountPerFilterValue>
+  >;
+} & Partial<Record<SmartTagFamily, number>>;
+
+const SmartTagsTable: React.FC<{
+  jobId: string;
+  pipeline: Required<QueryPipelineState>;
+  availableDatasetSplits: AvailableDatasetSplits | undefined;
+}> = ({ jobId, pipeline, availableDatasetSplits }) => {
+  const [datasetSplitName, setDatasetSplitName] =
+    React.useState<DatasetSplitName>("eval");
+  const [transpose, setTranspose] = React.useState(false);
+
+  const [selectedMetricPerFilterOption, setSelectedMetricPerFilterOption] =
+    React.useState<"label" | "prediction">("label");
+
+  const [ascending, setAscending] = React.useState(false);
+  const [sortBy, setSortBy] = React.useState<SortBy>("utteranceCount");
```

Comment:
@lindsaydbrin

> I noticed that when you switch between the training and eval sets, the order of the classes changes.

The table is initially sorted by the total utterance count column. I chose that to be consistent with the performance analysis table, which is just above this one in the dashboard, but we can change that here. Of course, the utterance counts change between the dataset splits.

If you sort by label/prediction classes, then the order persists between the dataset splits. Maybe that should be the default order when you load the page? So that if your first interaction is changing the dataset split, the order persists. What do you think?
```suggestion
  const [sortBy, setSortBy] = React.useState<SortBy>("filterValue");
```

Reply:
Oh, interesting. I wasn't sure about the default because I'd already been sorting things, and then I was startled to see the order change. Like, if you sort by accuracy and see a class you're interested in, you might want to have it stay put when you switch to the other dataset split.

But that said, I think it's reasonable to have the same behavior as the performance analysis table (default/initial view as well as switching sets). So maybe my suggestion/use case is a design question that we can think about later.